### PR TITLE
feat(vscode): completions for `bring` and hiding more irrelevant completions

### LINKED ISSUE #3975: Broken IntelliSense for `bring` statement
### I tried this:

1) When typing:
```
bring
```
I expect to see the entire list of submodules available in the Wing SDK.

Instead I get nothing:

bring1

2) When typing:
```
bring cl
```
I expect to see `cloud` as suggested autocompletion.

Instead I get nothing:

bring2

3) When typing:
```
bring cloud
```

I can finally get the IntelliSesese autocompletion (but it's not useful at this point):

bring3

### This happened:

No IntelliSense autocompletion for `bring` statements.

### I expected this:

I expected IntelliSense to provide autocompletion.

### Is there a workaround?

_No response_

### Component

IDE Extension

### Wing Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### LINKED ISSUE #3991: vscode suggests an illegal assert completion
### I tried this:

image

### This happened:

assert is autocompleted although it is not legal here

### I expected this:

not to complete to assert 

### Is there a workaround?

_No response_

### Component

IDE Extension

### Wing Version

0.26.5

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### LINKED ISSUE #3997: vscode annoying completion to inflight
### I tried this:

image

### This happened:

I think that the inflight completion is too eager

### I expected this:

to be smarter, but I am not sure how

### Is there a workaround?

_No response_

### Component

IDE Extension

### Wing Version

0.25.6

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by MarkMcCulloh**
##### Fixes #3991
image

##### Fixes #3997
image

##### Fixes #3975
image

_Didn't add documentation for them yet, wasn't sure how to go about it without some duplicated hardcoding._

---

Misc:
- `bring` followed by a single character previously caused a panic (e.g. `bring c`). Most people probably didn't notice cause they were typing `bring cloud` so fast
- No completions in comments or strings
- Tried to centralize many of the cases where it's obvious there will be no completions available. `completions.rs` is still begging for a rewrite.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingc/src/parser.rs [hunk 1]
```diff
@@ -791,15 +791,23 @@ impl<'s> Parser<'s> {
 	}
 
 	fn build_bring_statement(&self, statement_node: &Node) -> DiagnosticResult<StmtKind> {
-		let module_name_node = self.get_child_field(&statement_node, "module_name")?;
+		let Some(module_name_node) = statement_node.child_by_field_name("module_name") else {
+			return self.with_error("Expected built-in module or \"external module\"", &statement_node.child(statement_node.child_count() - 1).unwrap_or(*statement_node));
```
Comment (MarkMcCulloh) [inline libs/wingc/src/parser.rs:795]: Not totally happy with this wording, very open to suggestions

File: examples/tests/invalid/bring.test.w [hunk 2]
```diff
@@ -4,4 +4,8 @@ bring cloud;
 bring cloud;
 //    ^^^^^ "cloud" is already defined
 bring fs;
-//    ^^^^^ "fs" is not a built-in module
\ No newline at end of file
+//    ^^^^^ "fs" is not a built-in module
+bring ;
+//^^^^^ Expected built-in module or "external module"
```
Comment (eladb) [inline examples/tests/invalid/bring.test.w:9]: My plan is to reserve the non quoted namespace for "managed" modules from the `winglang/library` repository. See https://github.com/winglang/wing/issues/1037 (copy @Chriscbr).

So "built-in" is likely not the right term. Also, this syntax has a variant of a local module as well 🤦‍♀️

Maybe:

```suggestion
//^^^^^ Expected module specification (see docs/modules).
```

File: libs/wingc/src/lsp/completions.rs [hunk 3]
```diff
@@ -1485,4 +1567,74 @@ let x: Outer = { bThing: {   } }
...
+		r#"
+bring "" as 
+	        //^
+"#,
+		assert!(bring_alias.is_empty())
+	);
+
+	test_completion_list!(
+		definition_identifier_partial,
+		r#"
```
Comment (eladb) [inline libs/wingc/src/lsp/completions.rs:1598]: These tests are fabulous

File: libs/wingc/src/parser.rs (hunk 1 above)
Comment (eladb) [reply libs/wingc/src/parser.rs:795]: See my suggestion above although not sure if it's better.

**Review by eladb — APPROVED** [review]

**Pull request merged**